src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -1281,6 +1258,26 @@ class RideHailManager(
     }
   }
 
+  private def getActualResponses(request: RideHailRequest) = {
+    val previousResponses = rideHailResponseCache.getOrElse(request.customer, IndexedSeq.empty)
+    val currentTime = request.requestTime.getOrElse(0)
+    previousResponses.filter(_.request.departAt >= currentTime)
+  }
+
+  private def removeOriginalResponseFromCache(request: RideHailRequest) = {
+    val actualResponses: IndexedSeq[RideHailResponse] = getActualResponses(request)
+    val departureTimeOrdering =
+      Ordering.by((rsp: RideHailResponse) => Math.abs(request.departAt - rsp.request.departAt))
+    val maybeOriginalResponse = actualResponses.reduceOption(departureTimeOrdering.min)
```

Comment:
If choosing the minimum, then `if(actualResponses.isEmpty) None else actualResponses.minBy(_.departAt)` seems a better option?

Reply:
In this case, we would have 2 branches instead of one.